xaitk_saliency/impls/vis_sal_similarity/similarityscoring.py

```diff
@@ -0,0 +1,90 @@
+from xaitk_saliency import ImageSimilaritySaliencyMapGenerator
+from xaitk_saliency.utils.masking import weight_regions_by_scalar
+
+import numpy as np
+from sklearn.preprocessing import minmax_scale
+from scipy.spatial.distance import cdist
+
+
+class SimilarityScoring (ImageSimilaritySaliencyMapGenerator):
+    """
+    This saliency implementation transforms proximity in feature
+    space into saliency heatmaps. This should
+    require a sequence of feature vectors of the query and
+    reference image, a number of feature vectors as predicted
+    on perturbed images, as well as the masks of the reference image
+    perturbations (as would be output from a
+    `PerturbImage` implementation.
+
+    The perturbation masks used by the following implementation are
+    expected to be of type integer. Masks containing values of type
+    float are rounded to the nearest value and binarized
+    with value 1 replacing values greater than or equal to half of
+    the maximum value in mask after rounding while 0 replaces the rest.
+
+    param proximity_metric: The type of comparision metric to be used
+        to determine proximity in feature space. The following
+        metrics are currently supported.
+
+        ‘braycurtis’, ‘canberra’, ‘chebyshev’, ‘cityblock’, ‘correlation’,
+        ‘cosine’, ‘dice’, ‘euclidean’, ‘hamming’, ‘jaccard’, ‘jensenshannon’,
+        ‘kulsinski’, ‘mahalanobis’, ‘matching’, ‘minkowski’, ‘rogerstanimoto’,
+        ‘russellrao’, ‘seuclidean’, ‘sokalmichener’, ‘sokalsneath’,
+        ‘sqeuclidean’, ‘wminkowski’, ‘yule’.
+    """
+
+    def __init__(
+        self,
+        proximity_metric: str = 'euclidean'
+    ):
+        self.proximity_metric: str = proximity_metric
```

Comment:
Is there any way we can check that this is a valid input at construction time with some referral to the utilized function/module? Worse case we could possibly do a very simple invocation to the function to check that this is a valid parameter for it.

Reply:
I created a dummy construction to check if proximity_metric is valid. An alternative would be to store all the supported strings, but I could see that cause tracking issues across versions, so I went with the current fix.

Reply:
:+1: